src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -806,10 +806,10 @@ private TypeSymbol GetTupleFieldType(BoundExpression expression, CSharpSyntaxNod
                         hasError = true;
                         Error(diagnostics, ErrorCode.ERR_FieldCantHaveVoidType, errorSyntax);
                     }
-                    else if (expressionType.IsRestrictedType())
+                    else if (expressionType.IsPointerType() || expressionType.IsRestrictedType())
```

Comment:
>expressionType.IsPointerType() [](start = 29, length = 30)

It feels like we have a bigger problem, I believe pointers and restricted types should be flagged by constraints check (ConstraintHelper.CheckConstraints). Since this doesn't happen, we probably not verifying constraints for tuple types at all. I think we should do that instead of trying to duplicate some of the logic here.